README.md

```diff
@@ -9,3 +9,65 @@ See [Declarative String Processing Overview][decl-string]
+
+| apple/swift branch  | apple/swift-experimental-string-processing branch     |
+| ------------------- | ----------------------------------------------------- |
+| main                | swift/main                                            |
+| release/5.7         | swift/release/5.7                                     |
+| ...                 | swift/...                                             |
+
+A pair of corresponding branches are expected to build successfully together and pass all tests.
+
+### Integration workflow
+
+To integrate the latest changes in apple/swift-experimental-string-processing to apple/swift, carefully follow the workflow: 
+
+- Create pull requests.
+  - Create a pull request in apple/swift-experimental-string-processing from `main` to `swift/main`, e.g. "[Integration] main -> swift/main".
+  - If apple/swift needs to be modified to work with the latest `main` in apple/swift-experimental-string-processing, create a pull request in apple/swift.
+- Trigger CI.
+  - In the apple/swift-experimental-string-processing pull request, trigger CI using the following command (replacing `<PR NUMBER>` with the apple/swift pull request number, if any):
+    ```
+    apple/swift#<PR NUMBER> # use this line only if there is an corresponding apple/swift PR
+    @swift-ci please test
+    ```
+  - In the apple/swift pull request (if any), trigger CI using the following command (replacing `<PR NUMBER>` with the apple/swift-experimental-string-processing pull request number):
+    ```
+    apple/swift-experimental-string-processing#<PR NUMBER>
+    @swift-ci please test
+    ```
+- Merge when approved.
+  - Merge the pull request in apple/swift-experimental-string-processing as a **merge commit**.
+  - Merge the pull request in apple/swift (if any).
+
+### Development notes
+
+Compiler integration can be tricky. Use special caution when developing `_MatchingEngine`, `_CUnicode` and `_StringProcessing` modules.
+
+- Do not change the names of these modules without due approval from compiler and infrastructure teams.
+- Do not modify the existing ABI between the regular expression parser and the Swift compiler unless absolutely necessary. 
+- Always minimize the number of lockstep integrations, i.e. when apple/swift-experimental-string-processing and apple/swift have to change together.
+- In `_StringProcessing`, do not write fully qualified references to symbols in `_CUnicode`, and always wrap `import _CUnicode` in a `#if canImport(_CUnicode)`. This is because `_CUnicode` is built as part of `_StringProcessing` with CMake.
+- In `_MatchingEngine`, do not write fully qualified references to `_MatchingEngine` itself. This is because `_MatchingEngine` is built as `ExperimentalRegex` in `SwiftCompilerSources/` with CMake. 
```

Comment:
And related, we should rename it to `RegexParse` or something.

Reply:
I agree, but I think it is better for this document to describe the current state. We can patch it once the name has changed.